pipeline_tools/create_analysis_metadata.py

```diff
@@ -0,0 +1,502 @@
+
+    Args:
+        analysis_workflow_id (str): UUID of the analysis workflow.
+        **kwargs: Non-required but optional fields for analysis_process core. e.g. process_name, process_description.
+
+    Returns:
+        analysis_process_core (dict): Dict containing process_core metadata required for analysis_process.
+    """
+    analysis_process_core = {'process_id': analysis_workflow_id}
+    for optional_key, optional_val in kwargs:
+        analysis_process_core[optional_key] = optional_val
+    return analysis_process_core
+
+
+def get_analysis_process_type():
+    """Creates process_type metadata for analysis_process.
+
+    This defines the type of process, which refers to the process_type_ontology. The only required key of the dict
+    should be 'text', and the value should be the name of a process type being used.
+
+    Returns:
+        analysis_process_type (dict): The process_type metadata for analysis_process.
+    """
+    analysis_process_type = {'text': 'analysis'}
+    return analysis_process_type
+
+
+def get_workflow_metadata(metadata_file):
+    """Load workflow metadata from a JSON file.
+
+    Args:
+        metadata_file (str): Path to file containing metadata json for the workflow.
+
+    Returns:
+        metadata (dict): A dict consisting of Cromwell workflow metadata information.
+    """
+    with open(metadata_file) as f:
+        metadata = json.load(f)
+    return metadata
+
+
+def get_workflow_tasks(workflow_metadata):
+    """Creates array of Cromwell workflow's task metadata for analysis_process.
+
+    Args:
+        workflow_metadata (dict): A dict representing the workflow metadata.
+
+    Returns:
+        sorted_output_tasks (list): Sorted array of dicts representing task metadata in the format required for
+                                    the analysis json.
+    """
+    calls = workflow_metadata['calls']
+
+    output_tasks = []
+    for long_task_name in calls:
+        task_name = long_task_name.split('.')[-1]
+        task = calls[long_task_name][0]
+
+        if task.get('subWorkflowMetadata'):
+            output_tasks.extend(get_workflow_tasks(task['subWorkflowMetadata']))  # recursively parse tasks
+        else:
+            runtime = task['runtimeAttributes']
+            out_task = {
+                'task_name'   : task_name,
+                'cpus'        : int(runtime['cpu']),
+                'memory'      : runtime['memory'],
+                'disk_size'   : runtime['disks'],
+                'docker_image': runtime['docker'],
+                'zone'        : runtime['zones'],
+                'start_time'  : task['start'],
+                'stop_time'   : task['end'],
+                'log_out'     : task['stdout'],
+                'log_err'     : task['stderr']
+            }
+            output_tasks.append(out_task)
+    sorted_output_tasks = sorted(output_tasks, key=lambda k: k['task_name'])
+    return sorted_output_tasks
+
+
+def get_format(path, extension_to_format):
```

Comment:
I suggest renaming this to `get_file_format` to make it clearer what this function is doing

Reply:
Will do!